com.unity.multiplayer.mlapi/Tests/Editor/IndexAllocatorTests.cs

```diff
@@ -0,0 +1,109 @@
+using NUnit.Framework;
+using UnityEngine;
+
+namespace MLAPI.EditorTests
+{
+    public class FixedAllocatorTest
+    {
+        [Test]
+        public void SimpleTest()
+        {
+            int pos;
+
+            var allocator = new IndexAllocator(20000, 200);
+            allocator.DebugDisplay();
+
+            // allocate 20 bytes
+            Assert.IsTrue(allocator.Allocate(0, 20, out pos));
+            allocator.DebugDisplay();
+            Assert.IsTrue(allocator.Verify());
+
+            // can't ask for the same index twice
+            Assert.IsFalse(allocator.Allocate(0, 20, out pos));
```

Comment:
I would make sure it actually wasn't allocated

Reply:
I'm not sure how to do that. 

The `pos` received in the original allocation **is**  the allocation. If anything extra had been allocated, `Verify` would fail, as it tests that the sum of allocated index sums to the total (`if (total != m_BufferSize)` ). Can you be more specific as to what you would like to see ?

Reply:
ok, I missed your check for !m_Slots[pos].Free in Verify, nevermind